### ISSUE TITLE
tsgo does not detect unreachable code

### ISSUE THREAD

**Issue by Kumneger0** [microsoft/typescript-go#941]
#### Reproduction Steps

1. Follow the setup instructions from this blog post to enable `tsgo` in VSCode(if you haven't already):  
   https://devblogs.microsoft.com/typescript/announcing-typescript-native-previews/

2. Create a `.ts` file and add the following code:

   ```ts
   function doSomethingAwesome() {
     throw new Error("we messed up");
     return [];
   }

#### Expected Behavior
The return statement should be reported as unreachable when it follows a throw

#### Actual Behavior
No unreachable code warning is shown.


#### Environment 

- **@typescript/native-preview version:** `7.0.0-dev.20250525.1`
- **Node.js version:** `v20.18.2`
- **OS:** Ubuntu 24.04 (if applicable)

**Comment by jakebailey** [comment]
I think it does detect the unreachable code, it's just that the diagnostics code isn't setting that info on its response.

**Comment by jakebailey** [comment]
Hah, no, we're missing a bunch of related diagnostic stuff.
